pkg/operator/controllers/configs/doc.go

```diff
@@ -0,0 +1,13 @@
+package configs
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// Configs controller allows to create custom config that would be applied
+// into the cluster.
+// The design is inspired by the workaround controller. The difference is
+// the focus on configuration instead of hotfixes.
+// It also allows for flexible applicable criteria checks since it is
+// not fixed on any single item. All checks are done in the Config itself.
+// The configs operator takes the cluster config as a main object.
+// Other related objects are added in individual configs.
```

Comment:
I'm not convinced that we should implement a generic config controller at this point.

I would probably start with a controller which does one thing: reconciles one specific kubeletconfig. If/when we see more use cases for this we can reconsider having a generic controller for configs.